### DISCUSSION TITLE
Merging with thorin v2

### DISCUSSION THREAD

**Discussion by madmann91** [AnyDSL/thorin#106] [Ideas]
Here are the (incremental) goals to merge Thorin 2:

- Short term:

1. LLVM backend
2. Tuple params/app node/refactoring/merging types-Def, make debug info a Def
3. Serialization
4. Debugging infra
5. Modelling PrimOps with functions

- Long term:

6. Arbitrary PE filters
7. Optimization passes
8. Automatic differentiation
9. Garbage collection

**Comment by leissa** [comment]
Started working on 1) in the `llvm_rewrite` branch.

**Comment by leissa** [comment]
* `1` is 90% done.
* `4` is also 80% done but is subject to change as soon as we start working on 2)
